detectors/resources/src/main/java/com/google/cloud/opentelemetry/detectors/GKEResource.java

```diff
@@ -49,9 +49,16 @@ public Attributes getAttributes() {
     AttributesBuilder attrBuilders = Attributes.builder();
     attrBuilders.putAll(gceAttributes);
 
-    attrBuilders.put(ResourceAttributes.K8S_NAMESPACE_NAME, envVars.get("NAMESPACE"));
+    attrBuilders.put(
+        ResourceAttributes.CLOUD_PLATFORM,
+        ResourceAttributes.CloudPlatformValues.GCP_KUBERNETES_ENGINE);
     attrBuilders.put(ResourceAttributes.K8S_POD_NAME, envVars.get("HOSTNAME"));
```

Comment:
IIRC, this will be truncated if the pod name is too long. The python detector recommends setting this also via downward API. Wdyt?

Reply:
Added this as an optional thing, updated tests.